### PR TITLE
Reload prevented when user has unsaved data

### PR DESCRIPTION
**Pull request by atarka**
Added a warning before reloading or navigating away from the page when user has unsaved data in post or comment editing textarea to prevent accidental loss of said data.

### CODE REVIEW

File: src/components/loss-preventor.jsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { Children, useEffect, useState } from 'react';
+
+const getChildrenValue = (children) =>
+  Children.map(children, (child) => child.props.value || '').join(';');
+
+function LossPreventor(props) {
+  const [initialValue] = useState(getChildrenValue(props.children));
```
Comment (davidmz) [inline src/components/loss-preventor.jsx]: Здесь разумнее использовать useMemo, раз нужно только значение, а не его изменение. useState тут бесполезен — да, он сохранит только начальное значение, но getChildrenValue всё равно будет дёргаться при каждом рендере.

Comment (atarka) [reply src/components/loss-preventor.jsx]: И действительно, я забыл, что оно евалится nonetheless :) Заменил.

**Comment by davidmz** [comment]
I am trying to use onbeforeunload in my own code now and noticed that it is not fire on regular routing transitions i. e. during the navigation inside FreeFeed site. Did you check your code at these transitions?

**Comment by atarka** [comment]
Alas no, only browser navigation events are handled now. I'll dig into router Prompt to see if that can be addressed.

**Comment by davidmz** [comment]
Bad news: we still use ReactRouter v3 here:)

**Comment by atarka** [comment]
Fixed the issue - now I also set a router hook on current route to prevent navigation if user has unsaved changes.

**Comment by davidmz** [comment]
Why you close it?

**Comment by atarka** [comment]
I noticed a component of same functionality in your recent pull request and reasoned we don't need two components for one job. Besides yours can be used to track unsaved attachments too.

**Pull request closed without merging**